middleware/packet-forward-middleware/e2e/upgrade_test.go

```diff
@@ -0,0 +1,174 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	sdkmath "cosmossdk.io/math"
+	upgradetypes "cosmossdk.io/x/upgrade/types"
+	cosmosproto "github.com/cosmos/gogoproto/proto"
+	"github.com/docker/docker/client"
+	"github.com/strangelove-ventures/interchaintest/v8"
+	"github.com/strangelove-ventures/interchaintest/v8/chain/cosmos"
+	"github.com/strangelove-ventures/interchaintest/v8/ibc"
+	"github.com/strangelove-ventures/interchaintest/v8/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	chainName   = "simapp"
+	upgradeName = "v2" // x/params migration
+
+	haltHeightDelta    = uint64(9) // will propose upgrade this many blocks in the future
+	blocksAfterUpgrade = uint64(7)
+
+	VotingPeriod     = "15s"
+	MaxDepositPeriod = "10s"
+)
+
+var (
+	// baseChain is the current version of the chain that will be upgraded from
+	// docker image load -i ../prev_builds/pfm_7_0_1.tar
+	baseChain = ibc.DockerImage{
+		Repository: "pfm",
+		Version:    "v7.0.1",
+		UidGid:     "1025:1025",
+	}
+
+	// make local-image
+	upgradeTo = ibc.DockerImage{
+		Repository: "pfm",
+		Version:    "local",
+	}
+)
+
+func TestPFMUpgrade(t *testing.T) {
```

Comment:
I do not think this file is run at all, it can probably be removed safely for anything which is not main branch

Reply:
good call out, there may be a bit of dead code at this point. may need to just do an audit of that while we rectify any issues discovered in the audit.